module/zfs/zfs_log.c

```diff
@@ -511,7 +511,7 @@ zfs_log_write(zilog_t *zilog, dmu_tx_t *tx, int txtype,
 	if (zilog->zl_logbias == ZFS_LOGBIAS_THROUGHPUT)
 		write_state = WR_INDIRECT;
 	else if (!spa_has_slogs(zilog->zl_spa) &&
-	    resid >= zfs_immediate_write_sz)
```

Comment:
OpenZFS appears to be using `>=`, see [here](https://github.com/openzfs/openzfs/blob/master/usr/src/uts/common/fs/zfs/zfs_log.c#L469).. Why don't we want to be consistent with that?

Reply:
The concern is that the original OpenZFS commit changed how `zfs_immediate_write_sz` is interpreted [here](https://github.com/zfsonlinux/zfs/commit/1b7c1e5ce90ae27d9bb1f6f3616bf079c168005c#diff-b812ce31aba834fc2ae0f19891c6e916L507).  The ZoL port of 7578 patch faithfully preserved that change in behavior,

What's unclear is if it was intended or accidental, and if intentional how was it justified.  I was hoping @amotin who authored the patch or one of the original reviewed could shed some light on this.  We may want to instead update the documentation to be consistent.

Reply:
Ah, I see. So, if we change it here, we should probably change it in OpenZFS too?

I'll approve, since the code change seems fine in isolation, but I can't really say if this is "correct" or not. Hopefully @amotin will be able to comment.

Reply:
> Ah, I see. So, if we change it here, we should probably change it in OpenZFS too?

Yes, I definitely think we should keep the OpenZFS and ZoL behavior consistent if we determine this the "correct" thing to do.

Reply:
As I remember it is quite a gray area without obvious answers, very much depending on specific hardware, pool configuration and write pattern.  I did many tests at the time.  I don't remember now why exactly I changed > to >= there, and also why it is different from dataset code (though there are some differences that may affect), but I don't think it was a typo.  But if somebody wish to take responsibility and reconsider it (change of the operator without changing the value will change behavior) -- be my guest.